lib/facter/resolvers/linux/networking.rb

```diff
@@ -56,6 +58,26 @@ def parse_ip_command_line(line, mtu_and_indexes)
             mtu_and_indexes[name] = { index: index, mtu: mtu }
           end
 
+          def duplex(interface_name)
+            # not all interfaces support this, wifi for example does not
+            unless interface_name.start_with?('lo')
+              plex = Facter::Util::FileHelper.safe_read("/sys/class/net/#{interface_name}/duplex", nil)
+              @fact_list[:interfaces][interface_name][:duplex] = plex.strip if plex
+            end
+          rescue StandardError
+            nil
```

Comment:
The `safe_read` should eliminate the need for the rescuing `StandardError` here and below (since it checks if the file is readable before trying to read it).

Reply:
I've dropped the offending bits.